File: javascript/darts/darts.js
function solve(x, y) {
  if (isNaN(x) || isNaN(y)) {
    return null;
  }

  const target = {
    inner: {
      radius: 1,
      score: 10
    },
    middle: {
      radius: 5,
      score: 5
    },
    outer: {
      radius: 10,
      score: 1
    }
  };

  const distanceFromCentre = Math.sqrt(x ** 2 + y ** 2);

  return distanceFromCentre <= target.inner.radius
    ? target.inner.score
    : distanceFromCentre <= target.middle.radius
    ? target.middle.score
    : distanceFromCentre <= target.outer.radius
    ? target.outer.score
    : 0;
}

export { solve };

File: java/darts/src/main/java/Darts.java
class Darts {

    private double x;
    private double y;

    Darts(double x, double y) {
      this.x = x;
      this.y = y;
    }

    int score() {

      int score = 0;
      double distanceFromCentre = Math.hypot(this.x, this.y);

      if (distanceFromCentre <= 1) {
        score = 10;
      } else if (distanceFromCentre <= 5) {
        score = 5;
      } else if (distanceFromCentre <= 10) {
        score = 1;
      }

      return score;
    }

}
